### ISSUE TITLE
how to handle Flutter_reactive_ble's on bluetooth module's disconnection

### ISSUE THREAD

**Issue by vontdeux** [PhilipsHue/flutter_reactive_ble#311]
**Is your feature request related to a problem? (please describe)**
sometime the bluetooth module do disconnect, and I wonder is there a way to handle this? Is there some kind of onDisconnection trigger provided by the library?

**Describe the solution you'd like**
sometime the bluetooth module do disconnect, and I wonder is there a way to handle this? Is there some kind of onDisconnection trigger provided by the library?

I Have been using this library for my project, I can connect to it within a single class now, following example of this https://github.com/wolfc01/flutter_reactive_ble_uart_example/blob/master/lib/main.dart

I had some problem with it like using the library across multiple widget/classes, (maybe it is just my limited knowledge of flutter, because I am still learning)

But, sometime the bluetooth module do disconnect, and I wonder is there a way to handle this? Is there some kind of onDisconnection trigger provided by the library?

the example given only using a button to disconnect, and it is always a nested widget/page, but what if the disconnect happens from outside of the app?(ie: bluetooth module lost of power)

what if i want to maintain connection, and still get to control the subscription cancellation

TLDR : 
1. Is there a onDisconnection handler by the library?
2. Is there a simpler example that uses the library across multiple page/classes? (more on the state management)

**Comment by remonh87** [comment]
Hey I would check our example app. 

https://github.com/PhilipsHue/flutter_reactive_ble/blob/bc030414a540c92b00dc363891c8f1013eaf807e/example/lib/src/ble/ble_device_connector.dart#L45

We do the same there  ike what you need for your use case
